assets/js/state/sagas/hosts.js

```diff
@@ -52,6 +52,7 @@ export function* deregisterHost({ payload }) {
   yield put(setHostDeregistering(payload));
   try {
     yield call(del, `/hosts/${payload.id}`);
+    payload.navigate?.('/hosts');
```

Comment:
I agree to remove the optional chaining and unify the flows, we need to redirect always even on the host list, it's a no-op for the router, and we avoid to pass or not pass parameters, we don't have types and this cause only confusion imho.

Unify the things, pass the navigate always and remove the optional chaining